### PR TITLE
onBeforeSave validate and show worning alert.

### PR DESCRIPTION
**Pull request by amar-c8y**
onBeforeSave validate and show worning alert if there is mandatory feild is exist.

### CODE REVIEW

File: asset-properties-widget/component/asset-properties-config/asset-properties-config.component.ts [hunk 1]
```diff
@@ -22,12 +24,19 @@ export class AssetPropertiesConfigComponent {
...
+        this.isAssetSelected = true;
     }
+
+    onBeforeSave(config: any): boolean {
+        if (!config.asset || !(config.properties.map(function(item) { return item.active}).indexOf(true) > -1)) {
+            this.isAssetSelected = config.asset ? true : false;
+          return false;
+        }
+        return true;
+    }
```
Comment (ghost) [inline asset-properties-widget/component/asset-properties-config/asset-properties-config.component.ts]: This function can be simplified as
```
onBeforeSave(config: any): boolean {
  this.isAssetSelected = Boolean(config.asset);
  return !!(config.properties && some(config.properties, 'active'));
}
```

File: asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts [hunk 2]
```diff
@@ -97,7 +98,7 @@ export class AssetPropertiesSelectorComponent {
       });
       this.assetPropertySelectorModalRef.content.savePropertySelection.subscribe((properties:IManagedObject[]) => {
         this.properties = this.properties.concat(this.removeSelectedProperties(properties))
-        this.contextDashboardService.formDisabled = false;
+        this.isPropertySelected = true;
```
Comment (ghost) [inline asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts]:
```suggestion
        this.isAtleastOnePropertySelected = true;
```

File: asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts [hunk 3]
```diff
@@ -154,10 +155,10 @@ export class AssetPropertiesSelectorComponent {
...
         this.properties.splice(removeIndex, 1);
         property['active'] = false;
-        if(this.properties.length < 1){
-          this.contextDashboardService.formDisabled = true;
-        }else if(this.asset && this.properties.every(({ active }) => active)) {
-          this.contextDashboardService.formDisabled = false;
+        if(this.properties.length > 0 && this.properties.every(({ active }) => active)){
+          this.isPropertySelected = true;
+        }else {
+          this.isPropertySelected = false;
```
Comment (ghost) [inline asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts]: why is every used to check if atleast one property is selected or not?

File: asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts [hunk 4]
```diff
@@ -142,10 +143,10 @@ export class AssetPropertiesSelectorComponent {
     }
 
     updateOptions(){
-      if(this.asset && this.properties.map(function(item) { return item.active}).indexOf(true) > -1){
-        this.contextDashboardService.formDisabled = false;
+      if(this.properties.map(function(item) { return item.active}).indexOf(true) > -1){
+        this.isPropertySelected = true;
       }else {
-        this.contextDashboardService.formDisabled = true;
+        this.isPropertySelected = false;
```
Comment (ghost) [inline asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts]: Can simplify this without using lodash as
```
updateOptions() {
 this.isPropertySelected = _.some(this.properties, 'active');
}

```

**Review by ghost — CHANGES_REQUESTED** [review]

File: asset-properties-widget/component/asset-properties-config/asset-properties-config.component.ts (hunk 1 above)
Comment (amar-c8y) [reply asset-properties-widget/component/asset-properties-config/asset-properties-config.component.ts]: incorporated the changes.

File: asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts (hunk 3 above)
Comment (amar-c8y) [reply asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts]: every remove I am checking is atleast one property is exist or not and that flag used to show worning alert.

File: asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts (hunk 4 above)
Comment (amar-c8y) [reply asset-properties-widget/component/asset-properties-config/asset-property-selector/asset-property-selector.component.ts]: incorporated.

**Review by ghost — APPROVED** [review]

**Review by spau6 — APPROVED** [review]

**Pull request merged**